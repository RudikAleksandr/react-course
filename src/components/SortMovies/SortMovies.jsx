import React from 'react';
import { Row, Dropdown, DropdownButton, ButtonGroup } from 'react-bootstrap';
import PropTypes from 'prop-types';
import cn from './SortMovies.module.css';

const sortTypes = ['name', 'rating', 'release date'];

const SortMovies = ({ activeType = 'name', onSelectSortType }) => {

  const sortTypeItems = sortTypes.map((sortType, index) => (
    <Dropdown.Item
      className={cn.sortTypeItem}
      key={index}
      eventKey={sortType}>
      {sortType}
    </Dropdown.Item>
  ));

  return (
    <Row className={`align-items-center justify-content-end ${cn.sortMovies}`}>
      <span className={cn.sortByLabel}>sort by</span>
      <DropdownButton
        onSelect={onSelectSortType}
        as={ButtonGroup}
        className={cn.sortTypeDropdown}
        variant=""
        title={activeType}
      >
        {sortTypeItems}
      </DropdownButton>
    </Row>
  );
}

SortMovies.propTypes = {
  activeType: PropTypes.string,
  onSelectSortType: PropTypes.func.isRequired
};

export default SortMovies;
