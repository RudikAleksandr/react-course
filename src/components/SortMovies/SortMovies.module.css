.sortMovies {}

.sortMovies .sortByLabel {
  color: #777;
  margin-right: 20px;
  font-size: 1.15rem;
  text-transform: uppercase;
}

.sortMovies .sortTypeDropdown button {
  margin-top: 2px;
  color: white;
  text-transform: uppercase;
}

.sortMovies .sortTypeDropdown button:focus,
.sortMovies .sortTypeDropdown button:active {
  box-shadow: none;
}

.sortMovies .sortTypeItem::first-letter {
  text-transform: uppercase;
}
