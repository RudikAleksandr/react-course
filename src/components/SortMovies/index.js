import SortMovies from './SortMovies.jsx';

export default SortMovies;
