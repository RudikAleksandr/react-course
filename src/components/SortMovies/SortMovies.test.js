
import { render } from '@testing-library/react';
import SortMovies from './SortMovies';

describe('SortMovies component', () => {
  test('render SortMovies snapshot', () => {
    const component = <SortMovies activeType="vote_average" onSelectSortType={() => { }} />;
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });
});