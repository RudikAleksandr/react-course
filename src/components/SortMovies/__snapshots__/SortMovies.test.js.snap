// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortMovies component render SortMovies snapshot 1`] = `
<DocumentFragment>
  <div
    class="align-items-center justify-content-end sortMovies row"
  >
    <span
      class="sortByLabel"
    >
      sort by
    </span>
    <div
      class="sortTypeDropdown dropdown btn-group"
      role="group"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="dropdown-toggle btn"
        type="button"
      >
        rating
      </button>
    </div>
  </div>
</DocumentFragment>
`;
